### PR TITLE
Dmeerx 717 - Add Value Sets to Hot Rule Set

### PR DESCRIPTION
**Pull request by mickohanlon23**
- Updated HOT ruleset pre-population in order to add newly updated value sets.
- Completed some refactoring of the CQL pre-population file structure.
- Updated HOT questionnaires and libraries to reflect newly added value sets.

NOTE: Not yet using the "Oxygen Saturation Laboratory Test" Value Set due to an issue we need to resolve with how we handle labs. [Instead, individual codes are still being used for the lab tests.]

### CODE REVIEW

**Comment by yuliumitre** [comment]
The followings are the candidates to use valueSet instead of hard-coded options in the Questionnaire:
1.  "Type of Order" to use ValueSet-request-intent

File: HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql [hunk 1]
```diff
@@ -82,37 +58,41 @@ concept "Intent Original Order": { "Intent Original Order code" } display 'Origi
...
+
+//Arterial Oxygen Codes
 define "Arterial_oxygen_saturation_Codes": { "59408-5" }
 define "Arterial_partial_pressure_of_oxygen_Codes": { "2703-7" }
 define "Arterial_oxygen_saturation_during_exercise_Codes": { "89276-0" }
 
-define "Hematocrit_lab_test_Codes": { "20570-8", "31100-1", "32354-3", "41654-5", "41655-2", "4544-3", "4545-0", "71829-6", "71830-4", "71832-0", "71833-8" }
-define "Hematocrit_Codes": {"31100-1"}
+//Hematocrit Code
+define "Hematocrit_lab_test_Codes": {"32354-3"}
```
Comment (ghost) [inline HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql]: Why did you remove all of the other hematocrit lab test codes and are now only using one?

File: HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql [hunk 2]
```diff
@@ -82,37 +58,41 @@ concept "Intent Original Order": { "Intent Original Order code" } display 'Origi
...
+
 // Other codes
-define "Immobilization_Codes": { "102491009", "129857008", "129859006", "160685001", "371632003", "8510008" }
+define "Immobilization_Codes": { "Immobilization" }
```
Comment (ghost) [inline HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql]: You Don't need to assign the Immobilization value set to a variable in this line.

Below, when you define "PatientMobile", replace what you have with:
define PatientMobile: not exists(Confirmed(ActiveOrRecurring([Condition: "Immobilization"])))

File: HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql [hunk 3]
```diff
@@ -9,39 +9,35 @@ include FHIRHelpers version '4.0.0' called FHIRHelpers
 // with guidance from https://www.cms.gov/Outreach-and-Education/Medicare-Learning-Network-MLN/MLNProducts/Downloads/Home-Oxygen-Therapy-ICN908804.pdf
```
Comment (ghost) [inline HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql:8]: Overall, this file could use some better documentation to show how sections on grouped. For example, add comments to indicate where valuesets are defined, helper functions are defined, etc.

File: HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql (hunk 1 above)
Comment (mickohanlon23) [reply HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql]: The original list of hematocrit codes was based off of this Value Set: [2.16.840.1.113762.1.4.1045.114](https://vsac.nlm.nih.gov/valueset/2.16.840.1.113762.1.4.1045.114/definition) (not one of our Value Sets). I spoke with Carmela, and she said that the only hematocrit value that was truly relevant in this context is Hematocrit of Arterial blood (32354-3). If you look at the HOT Lab Value Set from the spreadsheet [[2.16.840.1.113762.1.4.1219.106](https://vsac.nlm.nih.gov/valueset/2.16.840.1.113762.1.4.1219.106/definition)], it only includes this one hematocrit code.

File: HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql [hunk 4]
```diff
@@ -9,39 +9,38 @@ include FHIRHelpers version '4.0.0' called FHIRHelpers
...
+valueset "Laboratory Test for Home Oxygen Therapy or Respiratory Assist Device Evaluation": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.106'
+
+// Immobilization Codes
+valueset "Immobilization": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.17.4077.3.2006'
+
+// Oxygen Saturation Lab Test
+valueset "Oxygen Saturation Lab Test": 'http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113762.1.4.1219.132'
+
+// Type of Device Order Valueset
+valueset "RequestIntent": 'https://www.hl7.org/fhir/valueset-request-intent.html'
```
Comment (yuliumitre) [inline HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql]: If only used in questionnaire, we don't need to declare in CQL file.

Comment (yuliumitre) [reply HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql]: @mickohanlon23 can you pull master to your branch? Valueset-R4-request-intent.json is in master but not in your branch so the questionnaire is not using it.

Comment (mickohanlon23) [reply HomeOxygenTherapy/R4/files/HomeOxygenTherapyPrepopulation-0.1.0.cql]: I deleted the unused Value Sets and rebased onto master.

**Comment by yuliumitre** [comment]
The type of order is not shown as a drop down list and it might be related to how the "DeviceOrderType" is handled. Right now it is returning a string, it probably should return the code of the value set.

Screen Shot 2020-07-28 at 10 03 10 AM

You can look at an example from 
ClinicalImpression defined in
CDS-Library/HomeHealthServices/R4/files/HomeHealthServicesPlanOfCarePrepopulation-0.0.1.cql
and used in CDS-Library/HomeHealthServices/R4/resources/Questionnaire-R4-HomeHealthServicesPlanOfCare.json

**Review by yuliumitre — APPROVED** [review]
The changes look good to me.
